indra/llcharacter/llik.h

```diff
@@ -523,7 +551,11 @@ class Solver
     LLVector3 computeReach(S16 to_id, S16 from_id) const;
 
     // Add a Joint to the skeleton.
-    void addJoint(S16 joint_id, S16 parent_id, const LLVector3& local_pos, const LLVector3& bone, const Constraint::ptr_t& constraint);
+    void addJoint(
+            S16 joint_id,
+            S16 parent_id,
+            LLJoint* info_ptr,
+            const Constraint::ptr_t& constraint);
```

Comment:
addJoint() now takes a raw` LLJoint*`.  The `local_pos `and `bone `(and scale!) are extracted from the **LLJoint** instance.